Feature/advice

key-del-jeeinho opened the pull request:
#8 #9 에 명시된 기능 구현 완료
---
상세 구현 내용은 위의 issue link 참고 바랍니다
- 시작일 : 2021.08.11
- 종료일 : 2021.08.12
- 소요기간 : 2일
- 개발인원 : 1명
    - @key-del-jeeinho

key-del-jeeinho committed:
Login/Register Failure Reason 구현
 - LoginFailureException 에 Reason enum 추가
 - RegisterFailureException 에 Reason enum 추가
 - 로그인/회원가입시 발생하는 예외를 처리하는 Advice 에, 해당 Reason 을 통한 구체적인 Response 가 되도록 수정

key-del-jeeinho committed:
ErrorResponse 관련 작업 구현
- Login/Register FailureReason 을 Advice 에 적용
- ErrorResponse 및 ErrorStatus 구현
- ErrorResponse 및 ErrorStatus 를 Advice 에 적용

siwony commented on src/main/java/com/gitpolio/gitpolioserver/advice/LoginAdvice.java:
```diff
@@ -17,7 +19,25 @@
     }
 
     @ExceptionHandler
```
클래스 명시를 하지 않아도 해당 `ExceptionHandler`는 동작하나요?

key-del-jeeinho replied:
안그래도 발견해서 수정하고있었네요 ㅎㅎ 수정하여 PR 하겠습니다

siwony reviewed (changes requested):
확인했습니다.
comment에 대한 답변 달아주세요

key-del-jeeinho committed:
일부 누락된 코드 수정 및 리팩터링
- LoginAdvice 의 ExceptionHandler 의 handling exception 을 명시하지 않아 이를 수정
- Login / PasswordAdvice 에 대한 Test 코드 작성
- 이외 가독성 향상을 위한 코드 일부 수정
